library/core/src/array/mod.rs

```diff
@@ -36,8 +37,18 @@ pub use iter::IntoIter;
 /// // elements to produce is the length of array down there: only arrays of
 /// // equal lengths can be compared, so the const generic parameter `N` is
 /// // inferred to be 5, thus creating array of 5 elements.
-/// let array = core::array::from_fn(|i| i);
+///
+/// let array: [_; 5] = core::array::from_fn(|i| i);
```

Comment:
Given the big comment about type inference above this one, I think *this* example should stay of inferred width
```suggestion
/// let array = core::array::from_fn(|i| i);
```